Cron/Order.php

```diff
@@ -277,4 +287,28 @@ protected function addOrderComment(OrderInterface $order, string $status, string
 
         return $returnStatus;
     }
+
+    /**
+     * Get an Order from an order increment id. Attempts several retries due to Magento lag issues.
+     *
+     * @param string $orderIncrementId
+     *
+     * @return OrderInterface|null An order
+     */
+    protected function getOrderByIncrementId(string $incrementId) : ?OrderInterface
+    {
+        $orderFetchCount = 0;
+        do {
+            $order =  $this->orderHelper->getOrderByIncrementId($incrementId);
+            if ($order && $order->getId()) {
+                return $order;
+            }
+
+            $orderFetchCount++;
+            // phpcs:ignore
+            sleep(self::ORDER_FETCH_SLEEP_TIME);
+        } while ($orderFetchCount <= self::MAX_ORDER_FETCH_ATTEMPTS);
```

Comment:
This should be `<` rather than `<=` (or else, since you're starting at zero, it will end up doing 4 attempts even though the constant is set to 3).

Reply:
@davehensleyns8 Good point, for some reason I thought the increment would occur before the while evaluation at the end of the expression. Instead the expression is processed then compared. Thanks for calling it out.